server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserService.java

```diff
@@ -61,6 +66,14 @@ public void upsertUser(Username username, char[] password) throws UsersRepositor
         upsert(user, username, password);
     }
 
+    public boolean existUser(Username username) throws UsersRepositoryException {
+        try {
+            return usersRepository.contains(usersRepository.getUser(username.asMailAddress()));
```

Comment:
I'm not sure this is virtual hosting friendly. Do we have some tests regarding this?

IMO we should:

```
return  usersRepository.contains(username);
```